### PR TITLE
Add boats positions, zoom to boat, change round, set and display score

### PR DESCRIPTION
**Pull request by bartoz1**
image

### CODE REVIEW

File: lib/pages/race/moderator/event_statistics.dart [hunk 1]
```diff
@@ -1,21 +1,23 @@
...
   @override
-  Widget build(BuildContext context) {
+  Widget build(BuildContext context, WidgetRef ref) {
+    final currentRound = ref.watch(currentRoundProvider);
+    final currentRoundStatus = ref.watch(currentRoundStatusProvider);
+
+
     return Container(
-      height: height,
+      height: MediaQuery.of(context).size.height * 0.1,
```
Comment (MarekBrandt) [inline lib/pages/race/moderator/event_statistics.dart:20]: Is there a reason we should use media query instead of flex etc?

**Comment by MarekBrandt** [comment]
As I told you, I believe dots will be better that boats. However, the boats look cute

File: lib/providers/race_events.dart [hunk 2]
```diff
@@ -39,3 +42,71 @@ Stream<Map<String, List<int>>> teamScores(TeamScoresRef ref, String eventId) {
   });
   return controller.stream;
 }
+
+// TODO change to immutable map
```
Comment (MarekBrandt) [inline lib/providers/race_events.dart:46]: what about this?

File: lib/providers/race_events.dart [hunk 3]
```diff
@@ -39,3 +42,71 @@ Stream<Map<String, List<int>>> teamScores(TeamScoresRef ref, String eventId) {
...
+  TeamPositionNotifier(this.firebaseDb, String eventId, List<String> teams) : super(<String, LatLng>{}) {
+    final logger = getLogger("TeamPositionNotifier");
+    logger.i("initializing a TeamPositionNotifier provider");
+
+    for (final team in teams) {
+      final dbRef = firebaseDb.child('traces').child(eventId).child(team).child('lastPosition');
+      dbRef.onValue.listen((DatabaseEvent event) {
+        final data = event.snapshot.value as String;
+        logger.i("updating $team position: ${data.toString()}");
+        final latlong = data.split(', ');
```
Comment (MarekBrandt) [inline lib/providers/race_events.dart:59]: you don't have to do this by yourselft. The library (LatLng if i'm not mistaken) could accept it in this form

**Review by MarekBrandt — APPROVED** [review]
Even if you leave everything as it is, great PR

File: lib/pages/race/moderator/event_statistics.dart [hunk 4]
```diff
@@ -71,24 +73,33 @@ class EventStatistics extends StatelessWidget {
...
+                    const Text(
+                      "Round",
+                      style: TextStyle(
+                        color: Colors.black,
+                        fontSize: 20,
+                        fontWeight: FontWeight.bold,
+                      ),
+                    ),
+                    Text(
+                      "${currentRound} / 5",
```
[the remark this answers is not in the document]
Comment (bartoz1) [reply lib/pages/race/moderator/event_statistics.dart:91]: it is just temporary

File: lib/providers/race_events.dart (hunk 2 above)
Comment (bartoz1) [reply lib/providers/race_events.dart:46]: it should actually use an immutable map but it also works this way

File: lib/providers/race_events.dart [hunk 5]
```diff
@@ -39,3 +42,71 @@ Stream<Map<String, List<int>>> teamScores(TeamScoresRef ref, String eventId) {
...
+        var latitude = double.parse(latlong[0]);
+        var longitude = double.parse(latlong[1]);
+        var newMap = Map.of(state);
+        newMap[team] = LatLng(latitude, longitude);
+        state = newMap;
```
[the remark this answers is not in the document]
Comment (bartoz1) [reply lib/providers/race_events.dart:64]: that's how the riverpod "state" works here

File: lib/pages/race/moderator/event_statistics.dart (hunk 1 above)
Comment (bartoz1) [reply lib/pages/race/moderator/event_statistics.dart:20]: nope, I was just playing with this value

**Pull request merged**